src/snapred/backend/api/InterfaceController.py

```diff
@@ -1,33 +1,39 @@
+import inspect
+
 from snapred.backend.dao.SNAPRequest import SNAPRequest
 from snapred.backend.dao.SNAPResponse import SNAPResponse
 from snapred.backend.log.logger import snapredLogger
 from snapred.backend.service.ServiceFactory import ServiceFactory
 from snapred.meta.decorators.Singleton import Singleton
 
-logger = snapredLogger.getLogger(__name__)
-
 
 @Singleton
 class InterfaceController:
     serviceFactory = ServiceFactory()
 
     def __init__(self):
         # make a singleton instance if one doesnt exist
-        pass
+        self.logger = snapredLogger.getLogger(self.__class__.__name__)
 
     def executeRequest(self, request: SNAPRequest) -> SNAPResponse:
-        result = None
-        response = None
         # execute the recipe
         # return the result
         try:
-            result = self.serviceFactory.getService(request.path).orchestrateRecipe(request)
-
+            # ServiceFactory doesn't always return an instance so it needs to be fixed here
```

Comment:
I think the instantiation code should get extracted from ServiceDirectory to a class method and then propagated to its [get method](https://github.com/neutrons/SNAPRed/blob/next/src/snapred/backend/service/ServiceDirectory.py#L33)